indico/modules/events/abstracts/models/abstracts.py

```diff
@@ -564,13 +570,31 @@ def can_judge(self, user, check_state=False):
     def can_edit(self, user):
         if not user:
             return False
-        is_manager = self.event.can_manage(user)
-        if not self.user_owns(user) and not is_manager:
+
+        manager_edit_states = (
+            AbstractPublicState.under_review,
+            AbstractPublicState.withdrawn,
+            AbstractPublicState.awaiting,
+            AbstractPublicState.invited,
+        )
+        if self.public_state in manager_edit_states and self.event.can_manage(user):
+            return True
+        elif self.public_state not in (AbstractPublicState.awaiting, AbstractPublicState.invited):
             return False
-        elif is_manager and self.public_state in (AbstractPublicState.under_review, AbstractPublicState.withdrawn):
+        elif not self.user_owns(user) or not self.event.cfa.can_edit_abstracts(user):
+            return False
+
+        editing_allowed = self.event.cfa.allow_editing
+        author_type = next((x.author_type for x in self.person_links if x.person.user == user), None)
+        is_primary = author_type == AuthorType.primary
+        is_secondary = author_type == AuthorType.secondary
+        if user == self.submitter:
```

Comment:
Shouldn't this be ...?
```suggestion
        if self.user_owns(user):
```

Reply:
that would allow any linked person again. `user_owns` would fit in case of `editing_allowed == AllowEditingType.submitter_all` but since it's checked above it's not needed there

Reply:
That makes sense.